plutus-core/prelude/PlutusPrelude.hs

```diff
@@ -240,3 +244,7 @@ zipExact _ _           = Nothing
 unsafeFromRight :: (Show e) => Either e a -> a
 unsafeFromRight (Right a) = a
 unsafeFromRight (Left e)  = error $ show e
+
+-- | function recursively applied N times
+timesA :: Natural -> (a -> a) -> a -> a
+timesA = ala Endo . stimes
```

Comment:
I've done some benchmarking a while ago and it was actually not a good way to define this function. The issue is that in

```haskell
instance Semigroup (Endo a) where
        (<>) = coerce ((.) :: (a -> a) -> (a -> a) -> (a -> a))
        stimes = stimesMonoid
```

`stimesMonoid` is defined like this:

```haskell
stimesMonoid :: (Integral b, Monoid a) => b -> a -> a
stimesMonoid n x0 = case compare n 0 of
  LT -> errorWithoutStackTrace "stimesMonoid: negative multiplier"
  EQ -> mempty
  GT -> f x0 n
    where
      f x y
        | even y = f (x `mappend` x) (y `quot` 2)
        | y == 1 = x
        | otherwise = g (x `mappend` x) (y `quot` 2) x               -- See Note [Half of y - 1]
      g x y z
        | even y = g (x `mappend` x) (y `quot` 2) z
        | y == 1 = x `mappend` z
        | otherwise = g (x `mappend` x) (y `quot` 2) (x `mappend` z) -- See Note [Half of y - 1]
```

which has quite a lot of bookkeeping that doesn't seem to benefit simple function composition at all. Not sure why they do it this way for function composition.

Reply:
I believe you that it does not have good performance. I am not currently relying on its performance, just its convenience for my testing. I can move it away from PlutusPrelude; let me know